polkadot/node/core/candidate-validation/src/lib.rs

```diff
@@ -773,7 +837,37 @@ async fn validate_candidate_exhaustive(
 					// invalid.
 					Ok(ValidationResult::Invalid(InvalidCandidate::CommitmentsHashMismatch))
 				} else {
-					Ok(ValidationResult::Valid(outputs, (*persisted_validation_data).clone()))
+					let core_index = candidate_receipt.descriptor.core_index();
+
+					match (core_index, exec_kind) {
+						// Core selectors are optional for V2 descriptors, but we still check the
+						// descriptor core index.
+						(
+							Some(_core_index),
+							PvfExecKind::Backing | PvfExecKind::BackingSystemParas,
```

Comment:
> Consider backing grp with 5 backers. 4 of the 5 are malicious. We need 3 of 5 for backing quorum.
> 
> Imagine backers produce a candidate with an invalid session index or mismatched core index. It will get backed because of the malicious coordinated backing group but when they distribute the candidate to the one honest backer will decide it is explicitly invalid.
> 

The runtime also checks the core and session index so such candidates are never backed.